Can't get ameba to compile on macOS 12.6

Hey y'all, I'm struggling to get ameba to compile on an M1 Mac. I've installed `crystal` through `brew`. This is my output when I run `make`.

```
➜  ameba git:(master) make
shards build -Dpreview_mt
Dependencies are satisfied
Building: ameba
Error target ameba failed to compile:
Undefined symbols for architecture arm64:
  "_EVP_MD_size", referenced from:
      _*Digest::SHA1@OpenSSL::Digest#digest_size:Int32 in D-igest5858S-H-A-1.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of command failed with code: 1: `cc "${@}" -o /Users/tjbarber/Projects/ameba/bin/ameba  -rdynamic -L/opt/homebrew/lib -L/usr/lib `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libcrypto || printf %s '-lcrypto'` -L/opt/homebrew/Cellar/libyaml/0.2.5/lib -lyaml -lpcre -lgc -L/opt/homebrew/Cellar/libevent/2.1.12/lib -levent_pthreads -levent -L/opt/homebrew/Cellar/libevent/2.1.12/lib -levent -liconv`

make: *** [bin/ameba] Error 1
```

This is my `crystal env`:

```
CRYSTAL_CACHE_DIR=/Users/tjbarber/.cache/crystal
CRYSTAL_PATH=lib:/opt/homebrew/Cellar/crystal/1.6.1/libexec/../share/crystal/src
CRYSTAL_VERSION=1.6.1
CRYSTAL_LIBRARY_PATH=/opt/homebrew/lib:/usr/lib
CRYSTAL_OPTS=''
```

Here is my `brew list`, just in case I'm missing a dependency (I don't think I am, this seems like a linking issue with openssl?):

```
➜  ameba git:(master) brew list
==> Formulae
apr			ctags			gradle@6		libpthread-stubs	libyaml			openjdk			readline		xorgproto
apr-util		elasticsearch-full	graphite2		libtiff			little-cms2		openjdk@11		redis			xz
autoconf		fontconfig		harfbuzz		libtool			llvm@11			openssl@1.1		ruby-build		z3
automake		freetype		heroku			libx11			llvm@14			openssl@3		screenresolution	zstd
bat			fzf			heroku-node		libxau			lolcat			pcre			six
bdw-gc			gdbm			icu4c			libxcb			lz4			pcre2			sqlite
ca-certificates		gettext			jpeg-turbo		libxdmcp		lzo			pixman			subversion
cairo			giflib			libevent		libxext			m4			pkg-config		the_silver_searcher
coreutils		glib			libiconv		libxml2			mpdecimal		python@3.10		unixodbc
crystal			gmp			libpng			libxrender		neofetch		rbenv			utf8proc

==> Casks
brave-browser	discord		firefox		iterm2		macvim		ngrok		postico		vlc		vscodium
```

--- a/src/ameba/runner.cr
+++ b/src/ameba/runner.cr
@@ -1,5 +1,3 @@
-require "digest"
-
 module Ameba
   # Represents a runner for inspecting sources files.
   # Holds a list of rules to do inspection based on,
@@ -186,7 +184,7 @@ module Ameba
       # Keep track of the state of the source. If a rule modifies the source
       # and another rule undoes it producing identical source we have an
       # infinite loop.
-      processed_sources = [] of String
+      processed_sources = [] of UInt64
 
       # It is possible for a rule to keep adding indefinitely to a file,
       # making it bigger and bigger. If the inspection loop runs for an
@@ -207,7 +205,7 @@ module Ameba
     # Check whether a run created source identical to a previous run, which
     # means that we definitely have an infinite loop.
     private def check_for_infinite_loop(source, corrected_issues, processed_sources)
-      checksum = Digest::SHA1.hexdigest(source.code)
+      checksum = source.code.hash
 
       if loop_start = processed_sources.index(checksum)
         raise InfiniteCorrectionLoopError.new(

